synapse/handlers/search.py

```diff
@@ -63,11 +63,15 @@ def get_old_rooms_from_upgraded_room(self, room_id):
         while True:
             try:
                 predecessor = yield self.store.get_room_predecessor(room_id)
-                predecessor_room_id = predecessor["room_id"]
+                if not predecessor:
+                    # This room does not have a predecessor
+                    break
             except NotFoundError:
-                # This room does not have a predecessor
+                # Unknown room_id
```

Comment:
this means that `room_id` is not a known room. it doesn't mean that the predecessor room is known.